flake.nix

```diff
@@ -22,100 +22,102 @@
       privateFlake = loadPrivateFlake ./dev/private;
 
       inputs = privateFlake.inputs // publicInputs;
+
+      # Just the NixOS and Darwin modules
+      modules = import ./.;
     in
-    inputs.flake-parts.lib.mkFlake { inherit inputs; } {
-      imports = [
-        inputs.git-hooks-nix.flakeModule
-        inputs.treefmt-nix.flakeModule
-      ];
+    # builtins.getFlake is not available in flake-compat mode
+    if builtins ? getFlake then
+      inputs.flake-parts.lib.mkFlake { inherit inputs; } {
```

Comment:
Actually, Can we remove inputs.flake-parts from here? Because it introduces a hidden nixpkgs that we cannot override. Here is an example how this can be separated: https://github.com/NixOS/nix/blob/3b76d01f3be01792dc45a89fc4266a605494ed84/flake.nix#L80

Reply:
Someone did already contact me about that not long ago.